query/src/sessions/context_shared.rs

```diff
@@ -58,6 +59,7 @@ pub struct DatabendQueryContextShared {
     pub(in crate::sessions) ref_count: Arc<AtomicUsize>,
     pub(in crate::sessions) subquery_index: Arc<AtomicUsize>,
     pub(in crate::sessions) running_query: Arc<RwLock<Option<String>>>,
+    pub(in crate::sessions) http_query: Arc<RwLock<Option<HttpQueryHandle>>>,
```

Comment:
what's the major difference between the HttpQuery and the queries from mysql & clickhouse ? q.q

Reply:
for now:  http_query may exist after the session is stopped because we now have no other place to hold the results.

but you remend of another choice:

IF:

1.  we keep server-side sessions for clients
2. Keeping the session after the query stopped does not cost too much.

then we can 

1. use (session ID，query_seq) as request handle for async query to find itself.
2. make struct http_query a optional part of a session.

Reply:
what will happen on we started a HTTP query and it is running, then we executed a `KILL <qid>` in the mysql console? q.q